src/providers/kafka-service.ts

```diff
@@ -0,0 +1,97 @@
+import { Component } from '@angular/core'
+import { NavController } from 'ionic-angular'
+import { Injectable } from '@angular/core'
+import { Observable } from 'rxjs/Observable'
+import 'rxjs/add/operator/map'
+import { Http, Response } from '@angular/http'
+
+import * as KafkaRest from 'kafka-rest/lib/client'
+var Kafka = new KafkaRest({ 'url': 'https://radararmt.ddns.net/kafka' }) //change to production or development url 
+
+
+@Injectable()
+export class KafkaService {
+
+ private aRMTIdSchema:string;
+ private aRMTInfoSchema:string;
+ 
+ private url = 'assets/data/schema/schemas.json'
+ private phq8schema : any;
+
+  constructor (
+    private http: Http
+  ) {
+  }
+
+
+
+ submitQuestionare(values):Promise<any>
+ {
+    return new Promise((resolve,reject) => {
+
+    this.getSchema().subscribe(
+    function(schema) { this.phq8schema = schema },
+    function(error) { console.log("Error at" + error)},
+    function() {
+
+      this.aRMTIdSchema = new KafkaRest.AvroSchema({"type": 'int'})
+      this.aRMTInfoSchema = new KafkaRest.AvroSchema(this.phq8schema)
+
+      Kafka.topic("active_questionnaire_phq8").produce(this.aRMTIdSchema,this.aRMTInfoSchema,
+      {'key':1,'value': {
```

Comment:
Since each questionnaire has its proper schema, each questionnaire will have a proper topic. A Kafka topic supports a single schema for the value. 

Each topic should have [MeasurementKey](https://github.com/RADAR-CNS/RADAR-Schemas/blob/master/commons/key/measurement_key.avsc) as key and the proper questionnaire schema as value.

The used questionnaire schema should be listed in the `RADAR-Schemas` repository under `questionnaire` folder as `.avsc` file.